src/rgw/driver/rados/rgw_rest_bucket.cc

```diff
@@ -221,6 +221,14 @@ void RGWOp_Bucket_Remove::execute(optional_yield y)
   std::unique_ptr<rgw::sal::Bucket> bucket;
 
   RESTArgs::get_string(s, "bucket", bucket_name, &bucket_name);
+  // parse tenant from bucket name (tenant:bucket)
+  std::string tenant;
+  auto pos = bucket_name.find(':');
+  if (pos != std::string::npos) {
+    tenant = bucket_name.substr(0, pos);
+    bucket_name = bucket_name.substr(pos + 1);
+  }
```

Comment:
can we just add a separate `tenant` param? then we won't a) rely on string parsing, or b) change the semantics of `bucket`